pkg/kubelet/kubelet_test.go

```diff
@@ -454,7 +452,9 @@ func TestSyncPodsDeletesWhenSourcesAreReadyPerQOS(t *testing.T) {
 	kubelet.HandlePodCleanups()
 	kubelet.HandlePodCleanups()
 	kubelet.HandlePodCleanups()
-	time.Sleep(2 * time.Second)
+
+	testKubelet.Cleanup()
+	testKubelet.kubelet.podKiller.Close()
```

Comment:
Is there a way to defer these for just the block above? With this change, if something explodes in the code above these won't get called, right?

Reply:
The test doesn't return early... so these should always get called.